cpp/ps0004/problem 5.cpp

```diff
@@ -1,26 +1,34 @@
-# include <iostream>
+#include <iostream>
 using namespace std ;
 int main ()
 {
-float number; 
+    // prompting user for number 1.5/1.5
+    float number; 
     cout << "Enter a number (must be between 32 and 126): ";
     cin >> number;
-    if ( number - int (number) != 0)
-  {  // إذا كان المدخل غير صحيح(رقم عشري)
+
+    // checking invalidaty of user's input 4/4
+	// printing the number and the opposite printable character 2/2
+    if (number - int(number) != 0)
+    {  // إذا كان المدخل غير صحيح(رقم عشري)
         cout << "Invalid: a floating point number " << endl;
     }
     
-   else if (number == 0 || cin.fail() ){
+    else if (number == 0 || cin.fail() ){
         // إاذا كان المدخل غير صحيح(مثلا حرف أو صفر)
         cout << "Invalid: character or zero value" << endl;
-        }
-      else if ( number < 32 || number > 126 || number < 0)
-      {//إذا كان المدخل سالب او خارج الرينج
-      cout << "Invalid number is out of the range" << endl;   }
-       else {
+    }
+    
+    else if ( number < 32 || number > 126)
+    {//إذا كان المدخل سالب او خارج الرينج
+        cout << "Invalid number is out of the range" << endl;
+    }
+    
+    else {
         // طباعة الرمز المقابل في جدول ASCII
         cout << "The character for " << number << " is " << char(number) << endl;
     }
```

Comment:
- [x] used `if` structure in a right way
- [x] `if` structure take in consideration all invalid cases
- [x] all expressions to check invalidity are correct
- [x] used `std::cout` and `std::endl` or `\n` to print a different message for the 3 cases
- [x] converted the number to the opposite printable character
- [x] printed the number and the opposite printable character